fcrepo-exts/fcrepo-camel-toolbox
# Fcrepo-3746: Docker improvements

**The title of this pull-request should be a brief description of what the pull-request fixes/improves/changes. Ideally 50 characters or less.**
* * *

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3746

* This PR is based on the PR from @mikejritter: https://github.com/fcrepo-exts/fcrepo-camel-toolbox/pull/163

# What does this Pull Request do?

* Simplifies the Dockerfile and docker-compose setup by not depending on a mounted configuration file and instead using environment variables.
* Adds the necessary steps to the Github Actions pipeline to build the docker image and push it to docker hub.

# What's new?

* Removes hardcoded path to configuration file, which gives users the ability to decide where they wan't to load the configuration from
* Fixes remaining issues in docker compose setup by setting fcrepo.authHostName (used in LDPath service) to `fcrepo` as well
* Uses `fcrepo` as collection name in Solr and database name in Fuseki
* Adds pipeline instructions to build and push the docker image

# How should this be tested?

To test locally:
* Run the maven build locally:<br/>`mvn clean package`
* Build the docker image:<br/>`docker build --tag="fcrepo/fcrepo-camel-toolbox:latest" --tag="fcrepo/fcrepo-camel-toolbox:6.0.0-SNAPSHOT" .` to build the docker image
* Start fcrepo, solr, fuseki and the camel-toolbox in docker containers:<br/>`docker-compose up`
* Open a separate terminal, run the following command to easily follow the logs of the camel-toolbox:<br/>`docker logs -f fcrepo-camel-toolbox_camel-toolbox_1` 
* Create some resources in fedora (accessible on http://localhost:8080/fcrepo/rest)
* Query Solr:<br/>`curl 'http://localhost:8983/solr/fcrepo/select?_=1633362335425&q=*:*&q.op=OR' -H 'Accept: application/json, text/plain, */*'`<br/>--> You should see the newly created resources in the response.
* Query Fuseki:<br/>`curl 'http://localhost:3030/fcrepo/query' -X POST -H 'Accept: application/sparql-results+json,*/*;q=0.9' --compressed -H 'Content-Type: application/x-www-form-urlencoded; charset=UTF-8' --data-raw 'query=%0A%0ASELECT+%3Fsubject+%3Fpredicate+%3Fobject%0AWHERE+%7B%0A++%3Fsubject+%3Fpredicate+%3Fobject%0A%7D%0ALIMIT+25'`<br/>--> You should see the newly created resources in the response.


# Additional Notes:
Open questions from https://github.com/fcrepo-exts/fcrepo-camel-toolbox/pull/163 which are not resolved by this PR:

Example:
* Volumes we might want to expose

# Interested parties
@mikejritter, @fcrepo/committers
